src/registry.jl

```diff
@@ -381,8 +381,23 @@ function detecthdf5(io)
     end
     false
 end
+function detect_mudata(io)
+    seekstart(io)
+    if String(read(io, 6)) != "MuData"
+        return false
+    end
+    seekstart(io)
+    return detecthdf5(io)
+end
```

Comment:
Make it all bytes comparison to reduce overhead.

```suggestion
const MUON_MAGIC = UInt8['M', 'u', 'D', 'a', 't', 'a']
function detect_mudata(io)
    seekstart(io)
    match(io, MUON_MAGIC) || return false
    seekstart(io)
    return detecthdf5(io)
end
```